USAGE.md

```diff
@@ -3,34 +3,80 @@
+  ```
+  *uses `docker.internal` if docker is going to be used and `dns.podman` if podman. This does not need modification unless another domain name is required.*
+* the realm domain for the freeipa server
+  ```bash
+  export IPA_REALM=${CONTAINER_DOMAIN^^}
+  ```
+
+Or create a `.env` file for `(docker|podman)-compose`
+```bash
+# Example for podman
+CONTAINER_RUNTIME=podman
+VAULT_LICENSE=02MV4UU..
+echo CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-docker} > .env
+if [[ -z ${CONTAINER_RUNTIME##docker} ]];then echo CONTAINER_DOMAIN=docker.internal; else echo CONTAINER_DOMAIN=dns.podman; fi >> .env
+echo VAULT_LICENSE=${VAULT_LICENSE} >> .env
+echo IPA_REALM=${CONTAINER_DOMAIN^^} >> .env
+```
+
+## 3. Start Containers
 
 Start FreeIPA, Keycloak, Vault and K3s:
 ```bash
-docker-compose up -d
+sudo ${CONTAINER_RUNTIME}-compose up -d
 ```
 
-## 3. Setup Resolution of Domain Names
+## 4. Setup Resolution of Domain Names
+
+You can either add a static record for every container to your `/etc/hosts` file or configure systemd-resolved service with `resolvectl` to use the container environment internal DNS (recommended).
+
+### Option 1: Configuration using container internal dns (recommended)
+First, find out which interface is used
+```bash
+CONTAINER_RUNTIME_IF=$(sudo ${CONTAINER_RUNTIME} network inspect vault-playground_vault-playground | jq -j '.[].network_interface')
+```
+which IP it has assigned
+```bash
+CONTAINER_RUNTIME_IF_IP=$(ip -j -p -br -4 addr show dev ${CONTAINER_RUNTIME_IF} | jq -j '.[].addr_info | .[].local')
+```
+and then configure this interface to resolve the domain specified
+```bash
+sudo -E resolvectl dns ${CONTAINER_RUNTIME_IF} ${CONTAINER_RUNTIME_IF_IP}
+sudo -E resolvectl domain ${CONTAINER_RUNTIME_IF} ${CONTAINER_DOMAIN}
+```
 
+### Option 2: Static configuration using hosts file
 Add a static A record for the container hosting FreeIPA, Keycloak and Vault. Alternatively, adjust your `/etc/hosts` file accordingly:
 
 ```bash
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} ipa.identity.net{{end}}' ipa) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} keycloak.identity.net{{end}}' keycloak) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} vault.identity.net{{end}}' vault) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} k3s-server.identity.net{{end}}' k3s-server) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} grafana.identity.net{{end}}' grafana) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} prometheus.identity.net{{end}}' prometheus) | sudo tee -a /etc/hosts
+echo -e $(sudo -E ${CONTAINER_RUNTIME} inspect -f "{{range .NetworkSettings.Networks}}{{.IPAddress}} ipa.${CONTAINER_DOMAIN}{{end}}" ipa) | sudo tee -a /etc/hosts
```

Comment:
Id suggests creating a script to add/remove/update those automatically.

Reply:
This is Option 2, wouldn't do that anyway :D Manipulating `/etc/hosts` is only an option for me if there is no other way. I'm bad at tidying up.

Reply:
> Id suggests creating a script to add/remove/update those automatically.

please add an issue if this is important